### DISCUSSION TITLE
About vim.system() and RPC

### DISCUSSION THREAD

**Discussion by brianhuster** [neovim/neovim#31175] [Q&A] [answered]
Hi, I just start learning about RPC, job and process in Neovim. When I take a look at the doc of `jobstart`, there is this line
```markdown
Note: Prefer |vim.system()| in Lua (unless using the `pty` option).
```
As I read the doc, `jobstart()` also has an option `rpc`, but `vim.system()` doesn't have one. Moreover, `vim.rpcrequest()` seems to only work with channels created by `jobstart()`. 

So I would like to ask if there is a way I can send an `rpcrequest` to a SystemObj created by `vim.system()`?

**Comment by justinmk** [comment, marked as answer]
True, `vim.system()` is missing `rpc` currently. We should enhance `vim.system()` to be the single interface for all such cases where invoking a system command is relevant. As opposed to `nvim_open_term` which doesn't involve a command invocation.

Would you mind creating an issue?

**Comment by brianhuster** [comment]
Ok, so the issue will be about missing RPC features in `vim.system()`, right? Btw `vim.system()` also doesn't support Vimscript callback, while `jobstart()` does, should I add it to the issue?

**Reply by lewis6991** [reply]
`vim.system` is purely Lua. If you want to invoke vimscript in the callback then you need to use `vim.cmd[[]]`.

**Reply by brianhuster** [reply]
I know that, I mean if I use vim.cmd or vim.fn in the callback, it will raise an error "Cannot use Vimscript function in Lua callback"

**Reply by zeertzjq** [reply]
`vim.schedule`
